bin/utils/cli-config-utils.js

```diff
@@ -55,7 +59,94 @@ function resolveDefaultKnexfilePath(extension) {
   return process.cwd() + `/knexfile.${extension}`;
 }
 
+function resolveEnvironmentConfig(opts, allConfigs) {
+  const environment = opts.env || process.env.NODE_ENV || 'development';
+  const result = allConfigs[environment] || allConfigs;
+
+  if (allConfigs[environment]) {
+    console.log('Using environment:', color.magenta(environment));
+  }
+
+  if (!result) {
+    console.log(color.red('Warning: unable to read knexfile config'));
+    process.exit(1);
+  }
+
+  if (argv.debug !== undefined) {
+    result.debug = argv.debug;
+  }
+
+  return result;
+}
+
+function exit(text) {
+  if (text instanceof Error) {
+    console.error(
+      color.red(`${text.detail ? `${text.detail}\n` : ''}${text.stack}`)
+    );
+  } else {
+    console.error(color.red(text));
+  }
+  process.exit(1);
+}
+
+function success(text) {
+  console.log(text);
+  process.exit(0);
+}
+
+function checkLocalModule(env) {
+  if (!env.modulePath) {
+    console.log(
+      color.red('No local knex install found in:'),
+      color.magenta(tildify(env.cwd))
+    );
+    exit('Try running: npm install knex');
+  }
+}
+
+function getMigrationExtension(env, opts) {
+  const config = resolveEnvironmentConfig(opts, env.configuration);
+
+  let ext = DEFAULT_EXT;
+  if (argv.x) {
+    ext = argv.x;
+  } else if (config.migrations && config.migrations.extension) {
+    ext = config.migrations.extension;
+  } else if (config.ext) {
+    ext = config.ext;
+  }
+  return ext.toLowerCase();
+}
+
+function getStubPath(env, opts) {
+  const config = resolveEnvironmentConfig(opts, env.configuration);
+  const stubDirectory = config.migrations && config.migrations.directory;
```

Comment:
this could be moved down. doesnt need to be assigned unless func reaches line 133